recipes/CommonVoice/self-supervised-learning/wav2vec2/train.py

```diff
@@ -0,0 +1,294 @@
+#!/usr/bin/env python3
+
+import sys
+import torch
+import logging
+import speechbrain as sb
+import torchaudio
+from hyperpyyaml import load_hyperpyyaml
+from speechbrain.utils.distributed import run_on_main
+
+"""Recipe for pretraining a wav2vec 2.0 model on CommonVoice EN. Note that it can be
+trained with ANY dataset as long as you provide the correct JSON or CSV file.
+
+The HuggingFace implementation of the wav2vec 2.0 pretraining is used and wrapped
+to fit properly the SpeechBrain framework. Models have been compared to the original
+fairseq implementation with success. The Transformers HuggingFace library is
+required:
+> pip install extra_requirements.txt
+
+Hence the process is the following:
+1. Indicate a HuggingFace repository that stores the wav2vec 2.0 config file.
+This is necessary to determine the architecture of the model that will be
+instantiated.
+2. Train it with our wrapper.
+3. Save it to be reused as a pretrained encoder within SpeechBrain (or others).
+
+wav2vec 2.0: https://arxiv.org/abs/2006.11477
+HuggingFace: https://huggingface.co/transformers/model_doc/wav2vec2.html
+
+To run this recipe, do the following:
+> python train.py hparams/hyperparams.yaml
+
+
+Authors
+ * Titouan Parcollet 2021
+ * Yan Gao 2021
+"""
+
+logger = logging.getLogger(__name__)
+
+
+# Define training procedure
+class W2VBrain(sb.core.Brain):
+    def compute_forward(self, batch, stage):
+        """Forward computations from the waveform batches to the w2v2 loss."""
+
+        batch = batch.to(self.device)
+        wavs, wav_lens = batch.sig
+        wavs, wav_lens = wavs.to(self.device), wav_lens.to(self.device)
+
+        # Normalize wav signal
+        wavs = self.modules.normalize(wavs, wav_lens)
+
+        # Forward on w2v2 and take the loss.
+        out, mask = self.modules.wav2vec2(wavs, mask_device=self.device)
```

Comment:
why you are not using the length ?
`self.model._get_feat_extract_output_lengths` need this information no ?

Reply:
I am not sure to understand properly.